### ISSUE TITLE
[rstream-query] unsubscribing one query makes other queries unusable

### ISSUE THREAD

**Issue by gavinpc-mindgrub** [thi-ng/umbrella#91]
First, thanks as usual for this fantastic project.  The `rstream-query` package has helped me to explore certain ideas much more productively.

My question is really this: is there a recommended way to do synchronous queries?  There are cases where I just want to get the results of a query one time.

To do that, I have used the following function:

```
const rq = require("@thi.ng/rstream-query");

const sync_query = (store, where) => {
  let results;
  const query = store
    .addQueryFromSpec({ q: [{ where }] })
    .subscribe({ next: result_set => (results = result_set) });
  query.unsubscribe();
  return results;
};

const store = new rq.TripleStore();
store.into([["apple", "isa", "fruit"], ["banana", "isa", "fruit"]]);
const results = sync_query(store, [["?thing", "isa", "fruit"]]);
console.log(`results`, results);
// Throws "illegal state: operation not allowed in DONE state"
// when trying to add the pattern query.
const results2 = sync_query(store, [["?thing", "isa", "fruit"]]);
console.log(`results2`, results2);
```

This works fine the first time.  However, when I run it a second time, I get the indicated "illegal state" error.

This appears to be the result of the sharing of streams representing common query clauses, combined with transitive unsubscription.  In fact, the above error occurs no matter what the second query is.  Here is an even simpler repro case

```
const rq = require("@thi.ng/rstream-query");

const store = new rq.TripleStore();
store.into([["apple", "isa", "fruit"], ["banana", "isa", "fruit"]]);

const query = store
  .addQueryFromSpec({ q: [{ where: [["?anything", "isa", "thing"]] }] })
  .subscribe({ next() {} });

query.unsubscribe();

// Throws
const query2 = store.addQueryFromSpec({
  q: [{ where: [["foo", "bar", "?baz"]] }]
});
```

In short, it seems that unsubscribing any query makes it impossible to add future queries, apparently due to the closure of the underlying shared streams.

I can avoid this by not unsubscribing the query.  But for one-time queries of course I don't want to leave dangling subscriptions.  And even for normal query subscriptions, it should be possible to remove these without affecting others.

Is this intended?  If not, I am happy to study the code further for a solution.

**Comment by postspectacular** [comment]
Hi @gavinpc-mindgrub - thank YOU! I will check this out more closely later, but at first glance this most likely is related to #74. Also see this comment: https://github.com/thi-ng/umbrella/issues/81#issuecomment-482062843 - more soon!
